lib/core/src/display/shape/text/text_field/word_occurrence.rs

```diff
@@ -180,6 +186,63 @@ fn get_word_occurrences(content:&[char], word:&[char]) -> Vec<Range<usize>> {
 #[cfg(test)]
 mod test {
     use super::*;
+    use crate::display::shape::text::text_field::content::test::mock_properties;
+
+    use basegl_core_msdf_sys as msdf_sys;
+    use wasm_bindgen_test::wasm_bindgen_test;
+
+    #[wasm_bindgen_test(async)]
+    async fn get_range() {
+        msdf_sys::initialized().await;
+        let content = "fn print_n_1(n:i32);\r\n#[test]\r\nfn test() {\r\n  print_n_1(1);\r\n}\n";
+        let content = TextFieldContent::new(content,&mock_properties());
+        let cursor  = Cursor::new(TextLocation { line:0, column: 4});
+        let range   = WordOccurrences::get_range(&content,&cursor);
+        assert_eq!(range, Some(TextLocation {line:0, column:3}..TextLocation{line:0, column:12}));
+
+        let cursor = Cursor::new(TextLocation { line:1, column: 0});
+        let range  = WordOccurrences::get_range(&content,&cursor);
+        assert!(range.is_none());
+
+        let mut cursor = Cursor::new(TextLocation { line:0, column: 4});
+        cursor.select_range(&(TextLocation {line:0, column:3}..TextLocation{line:0, column:12}));
+        let range = WordOccurrences::get_range(&content,&cursor);
+        assert_eq!(range, Some(TextLocation {line:0, column:3}..TextLocation{line:0, column:12}));
+    }
+
+    #[wasm_bindgen_test(async)]
+    async fn get_words() {
+        msdf_sys::initialized().await;
+        let content = "fn print_n_1(n:i32);\r\n#[test]\r\nfn test() {\r\n  print_n_1(1);\r\n}\n";
+        let content = TextFieldContent::new(content,&mock_properties());
+        let cursor  = Cursor::new(TextLocation { line:0, column: 4});
+        let range   = WordOccurrences::get_range(&content,&cursor);
+        let words   = WordOccurrences::get_words(&content,range).expect("Couldn't find words");
+        assert_eq!(words.len(), 2);
+
+        let words = WordOccurrences::get_words(&content,None);
+        assert_eq!(words,None);
+    }
+
+    #[wasm_bindgen_test(async)]
+    async fn new_word_occurrences() {
+        msdf_sys::initialized().await;
+        let content = "fn print_n_1(n:i32);\r\n#[test]\r\nfn test() {\r\n  print_n_1(1);\r\n}\n";
+        let content = TextFieldContent::new(content,&mock_properties());
+        let cursor  = Cursor::new(TextLocation {line:1,column:0});
+        let words   = WordOccurrences::new(&content,&cursor);
+
+        assert!(words.is_none());
+
+        let cursor    = Cursor::new(TextLocation {line:0,column:4});
+        let mut words = WordOccurrences::new(&content,&cursor).expect("Couldn't find words");
+
+        let word1 = WordRange::new(TextLocation{line:0,column:3}..TextLocation{line:0,column:12});
+        let word2 = WordRange::new(TextLocation{line:3,column:2}..TextLocation{line:3,column:11});
+        assert_eq!(words.select_next(), Some(word1));
+        assert_eq!(words.select_next(), Some(word2));
+        assert!(words.select_next().is_none());
```

Comment:
Add test where WordOccurrences::new would return None. Perhaps we want to test case for empty strings as well.

Reply:
None is already tested when there is no word occurrence. I have added a test for empty strings.